services-custom/s3-transfer-manager/src/main/java/software/amazon/awssdk/transfer/s3/DirectoryDownload.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.transfer.s3;
+
+import java.util.concurrent.CompletableFuture;
+import software.amazon.awssdk.annotations.SdkPreviewApi;
+import software.amazon.awssdk.annotations.SdkPublicApi;
+
+/**
+ * A download transfer of a directory of objects from S3
+ */
+@SdkPublicApi
+@SdkPreviewApi
+public interface DirectoryDownload extends DirectoryTransfer {
+    @Override
+    CompletableFuture<CompletedDirectoryDownload> completionFuture();
```

Comment:
Nit: Javadoc

Reply:
It has inherited Javadoc, do you think we still need to add more?
```
    /**
     * @return The future that will be completed when this transfer is complete.
     */
```